doc/BUILD.rst

```diff
@@ -0,0 +1,722 @@
+ignore them (``ignore``).
+
+Generally, ``no-error`` is the recommended default when using ``-Dwerror=true``.
+
+.. code:: sh
+
+   meson configure build/ -Ddeprecation_warnings=no-error
+
+
+Stack Trace Debugging (``-Db_sanitize``)
+""""""""""""""""""""""""""""""""""""""""
+
+This is useful for debugging crashes in the game. It uses
+`AddressSanitizer <https://github.com/google/sanitizers/wiki/AddressSanitizer>`__:
+
+.. code:: sh
+
+   meson configure build/ -Db_sanitize=address,undefined
+
+Depending on your platform, you may need to specify the specific library binary
+to use to launch ASan appropriately. Using macOS as an example:
+
+.. code:: sh
+
+   export DYLD_INSERT_LIBRARIES=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/12.0.0/lib/darwin/libclang_rt.asan_osx_dynamic.dylib
+
+The ``../12.0.0/..`` in the path of ``DYLD_INSERT_LIBRARIES`` changes with each
+version of Xcode. If it fails to launch for you, ensure that the version number
+is correct by browsing to the parent directory of ``../clang``.
+
+Then, you can launch Taisei's binary from the command line (using macOS as an
+example):
+
+.. code:: sh
+
+   /path/to/Taisei.app/Contents/MacOS/Taisei
+
+Link-Time Optimizations (``-Db_lto``)
+"""""""""""""""""""""""""""""""""""""
+
+* Default: ``true``
+* Options: ``true``, ``false``
+
+Link-time optimizations (LTO) increase build times, but also increase
+performance. For quicker build times during development, you can disable it.
+For release builds, this should be keep ``true``.
+
+See: `Interprocedural Optimization <https://en.wikipedia.org/wiki/Interprocedural_optimization#WPO_and_LTO>`__
+
+.. code:: sh
+
+   meson configure build/ -Db_lto=false
+
+Binary Striping (``-Dstrip``)
+"""""""""""""""""""""""""""""
+
+* Default: ``true``
+* Options: ``true``, ``false``
+
+This option prevents stripping of the `taisei` binary, leading to faster build
+times and keeping debugging symbols in place. There is a theoretical performance
+hit with this option enabled, but it can help with building during development.
+
+Keep this ``true`` during releases.
+
+.. code:: sh
+
+   meson configure build/ -Db_strip=false
+
+Rendering
+"""""""""
+
+Backends (``-Dr_gl*``)
+''''''''''''''''''''''
+
+* Default: ``false``
+* Options: ``true``, ``false``
+
+Enable or disable the various renderer backends for Taisei.
+
+.. code:: sh
+
+   # for GL 3.3 (default)
+   meson configure build/ -Dr_gl33=true
+   # for GL ES 3.0
+   meson configure build/ -Dr_gles30=true
+   # for GL ES 2.0
+   meson configure build/ -Dr_gles20=true
+
+Note that GL ES 2.0 requires a few extensions to be present on your system
+to function correctly, most notably:
+
+- ``OES_depth_texture`` or ``GL_ANGLE_depth_texture``
+- ``OES_standard_derivatives``
+- ``OES_vertex_array_object``
+- ``EXT_frag_depth``
+- ``EXT_instanced_arrays`` or ``ANGLE_instanced_arrays`` or
+  ``NV_instanced_arrays``
```

Comment:
1. We shouldn't encourage use of the gles20 backend; it is very undercooked, unmaintained, and basically useless at this point. The extensions it requires to work at all are really only available on gles3-capable hardware.
2. Probably worth noting here that shader translation is required for the gles3 backend to function properly.